pkg/lib/dialogs.jsx

```diff
@@ -115,10 +117,26 @@ export const WithDialogs = ({ children }) => {
                               JSON.stringify(dialog));
             is_open.current = !!component;
             setDialog(component);
+            return new Promise((resolve, reject) => {
```

Comment:
`Dialogs.show(null)` is legal and equivalent to `Dialogs.close()`... That might have been a bad idea, but I think we should continue to support it. Also, it is "allowed" to call `Dialogs.show()` twice in a row without any `Dialogs.close()` in between.

I guess all I'm trying to say is that `show` also needs to call the old `resolvePromise` before setting a new one.

Reply:
Or! We straight up change the API and declare it an error to call show() when a dialog is open and throw a direct exception back to the caller of the erroneous show().

Reply:
Or! We make a new function `Dialogs.interact` that is the one that can be awaited, and we don't change `Dialogs.show` at all.

Reply:
So, `Dialogs.show(nul)` can be used to close() that feels like a horrible API. Both methods are exposed already via `Dialogs`.

I am of the opinion that making `Dialogs.show(null)` deprecated is a good idea, and then just handling that in code. So adding a `console.warn`

Reply:
Or! We leave the code as it is in this PR and just drop the old promise on the floor and never resolve or reject it. We should in any case document it clearly that calling show() twice is a a bug, and if you want to replace a open dialog cleanly, you need to first explicitly close it. This makes a lot of sense now what dialogs can be awaited.